### ISSUE TITLE
2021.6.6-ls28 - Authentication broken

### ISSUE THREAD

**Issue by MunkeyBalls** [linuxserver/docker-homeassistant#13]
[linuxserverurl]: https://linuxserver.io
[![linuxserver.io](https://raw.githubusercontent.com/linuxserver/docker-templates/master/linuxserver.io/img/linuxserver_medium.png)][linuxserverurl]

------------------------------

## Expected Behavior
Being able to sign in to home assistant.

## Current Behavior
After updating from an older version (around 2021.6.1, not sure exactly which) I had to log in again, which is odd as normally homeassistant will remember your login. (I updated from a working config, rolling back to 2021.6.1 resolves the issue)

After logging in a message will appear that you are not able to connect to home assistant
![image](https://user-images.githubusercontent.com/24717620/122828440-fa15ab80-d2e5-11eb-814f-f9348f1e43be.png)

I have a feeling that [these ](https://github.com/home-assistant/core/issues/45128) [issues ](https://github.com/home-assistant/core/issues/46765)are related. Also, I have a feeling that this isn't the only thing that's broken. The reason why I forced an update was that my template sensors seemed to be not working. The template itself worked as tested from the developer menu, but the sensor itself didn't show correct state. Probably there are some incorrect packages installed, but ignore this for now. I'll retest and create a seperate issue whenever the auth issue is resolved.

## Steps to Reproduce

1. Update to one of the latest versions (if needed I can test exactly when this broke)
2. Start the container
3. Sign in
4. Get error

## Environment
**OS:** Debian 10
**CPU architecture:** arm64
**How docker service was installed:**

Docker is installed as listed in the docker install guide, added the docker sources and installed via apt-get install
Docker-compose is used to create / start the container

## Command used to create docker container (run/create/compose/screenshot)

```
homeassistant:
    image: linuxserver/homeassistant
    container_name: homeassistant
    network_mode: host
    environment:
      - PUID=1002
      - PGID=1002
      - TZ=Europe/Amsterdam
    volumes:
      - /home/homeassistant/homeassistant:/config
    restart: unless-stopped  
```
docker-compose up -d

## Docker logs

```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 01-envfile: executing... 
[cont-init.d] 01-envfile: exited 0.
[cont-init.d] 10-adduser: executing... 
-------------------------------------
          _         ()
         | |  ___   _    __
         | | / __| | |  /  \ 
         | | \__ \ | | | () |
         |_| |___/ |_|  \__/
Brought to you by linuxserver.io
-------------------------------------
To support LSIO projects visit:
https://www.linuxserver.io/donate/
-------------------------------------
GID/UID
-------------------------------------
User uid:    1001
User gid:    1001
-------------------------------------
[cont-init.d] 10-adduser: exited 0.
[cont-init.d] 50-config: executing... 
[cont-init.d] 50-config: exited 0.
[cont-init.d] 60-usb-gid: executing... 
[cont-init.d] 60-usb-gid: exited 0.
[cont-init.d] 90-custom-folders: executing... 
[cont-init.d] 90-custom-folders: exited 0.
[cont-init.d] 99-custom-files: executing... 
[custom-init] files found in /config/custom-cont-init.d executing
[custom-init] apk_add.sh: executing...
fetch http://dl-cdn.alpinelinux.org/alpine/v3.12/main/aarch64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.12/community/aarch64/APKINDEX.tar.gz
v3.12.7-79-ge9ed87cb7e [http://dl-cdn.alpinelinux.org/alpine/v3.12/main]
v3.12.7-57-g0c78d16d82 [http://dl-cdn.alpinelinux.org/alpine/v3.12/community]
OK: 12623 distinct packages available
**** installing python2 ****
fetch http://dl-cdn.alpinelinux.org/alpine/v3.12/main/aarch64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.12/community/aarch64/APKINDEX.tar.gz
(1/1) Installing python2 (2.7.18-r0)
Executing busybox-1.31.1-r20.trigger
OK: 180 MiB in 139 packages
[custom-init] apk_add.sh: exited 0
[custom-init] apk_add_arpscan.sh: executing...
[custom-init] apk_add_arpscan.sh: exited 0
[cont-init.d] 99-custom-files: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
2021-06-21 22:11:31 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration entity_controller which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2021-06-21 22:11:31 WARNING (SyncWorker_1) [homeassistant.loader] We found a custom integration shell_command which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2021-06-21 22:11:31 WARNING (SyncWorker_2) [homeassistant.loader] We found a custom integration eufy_vacuum which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2021-06-21 22:11:31 WARNING (SyncWorker_3) [homeassistant.loader] We found a custom integration cryptoinfo which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2021-06-21 22:15:48 ERROR (SyncWorker_4) [homeassistant.util.package] Unable to install package PyNaCl==1.3.0: ERROR: Command errored out with exit status 1:
   command: /usr/bin/python3 /usr/lib/python3.8/site-packages/pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-vg6dkv2y/overlay --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- setuptools wheel 'cffi>=1.4.1; python_implementation != '"'"'PyPy'"'"''
       cwd: None
  Complete output (166 lines):
  Collecting setuptools
    Using cached setuptools-57.0.0-py3-none-any.whl (821 kB)
  Collecting wheel
    Using cached wheel-0.36.2-py2.py3-none-any.whl (35 kB)
  Collecting cffi>=1.4.1
    Using cached cffi-1.14.5.tar.gz (475 kB)
  Collecting pycparser
    Using cached pycparser-2.20-py2.py3-none-any.whl (112 kB)
  Building wheels for collected packages: cffi
    Building wheel for cffi (setup.py): started
    Building wheel for cffi (setup.py): finished with status 'error'
    ERROR: Command errored out with exit status 1:
     command: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-avyx4ntk/cffi/setup.py'"'"'; __file__='"'"'/tmp/pip-install-avyx4ntk/cffi/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-pohvr3b7
         cwd: /tmp/pip-install-avyx4ntk/cffi/
    Complete output (68 lines):
    Package libffi was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libffi.pc'
    to the PKG_CONFIG_PATH environment variable
    Package 'libffi', required by 'virtual:world', not found
    Package libffi was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libffi.pc'
    to the PKG_CONFIG_PATH environment variable
    Package 'libffi', required by 'virtual:world', not found
    Package libffi was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libffi.pc'
    to the PKG_CONFIG_PATH environment variable
    Package 'libffi', required by 'virtual:world', not found
    Package libffi was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libffi.pc'
    to the PKG_CONFIG_PATH environment variable
    Package 'libffi', required by 'virtual:world', not found
    Package libffi was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libffi.pc'
    to the PKG_CONFIG_PATH environment variable
    Package 'libffi', required by 'virtual:world', not found
    unable to execute 'gcc': No such file or directory
    unable to execute 'gcc': No such file or directory
  
        No working compiler found, or bogus compiler options passed to
        the compiler from Python's standard "distutils" module.  See
        the error messages above.  Likely, the problem is not related
        to CFFI but generic to the setup.py of any Python package that
        tries to compile C code.  (Hints: on OS/X 10.8, for errors about
        -mno-fused-madd see http://stackoverflow.com/questions/22313407/
        Otherwise, see https://wiki.python.org/moin/CompLangPython or
        the IRC channel #python on irc.freenode.net.)
  
        Trying to continue anyway.  If you are trying to install CFFI from
        a build done in a different context, you can ignore this warning.
  
    running bdist_wheel
    running build
    running build_py
    creating build
    creating build/lib.linux-aarch64-3.8
    creating build/lib.linux-aarch64-3.8/cffi
    copying cffi/setuptools_ext.py -> build/lib.linux-aarch64-3.8/cffi
    copying cffi/api.py -> build/lib.linux-aarch64-3.8/cffi
    copying cffi/pkgconfig.py -> build/lib.linux-aarch64-3.8/cffi
    copying cffi/vengine_cpy.py -> build/lib.linux-aarch64-3.8/cffi
    copying cffi/cparser.py -> build/lib.linux-aarch64-3.8/cffi
    copying cffi/__init__.py -> build/lib.linux-aarch64-3.8/cffi
    copying cffi/verifier.py -> build/lib.linux-aarch64-3.8/cffi
    copying cffi/lock.py -> build/lib.linux-aarch64-3.8/cffi
    copying cffi/backend_ctypes.py -> build/lib.linux-aarch64-3.8/cffi
    copying cffi/error.py -> build/lib.linux-aarch64-3.8/cffi
    copying cffi/vengine_gen.py -> build/lib.linux-aarch64-3.8/cffi
    copying cffi/recompiler.py -> build/lib.linux-aarch64-3.8/cffi
    copying cffi/model.py -> build/lib.linux-aarch64-3.8/cffi
    copying cffi/cffi_opcode.py -> build/lib.linux-aarch64-3.8/cffi
    copying cffi/ffiplatform.py -> build/lib.linux-aarch64-3.8/cffi
    copying cffi/commontypes.py -> build/lib.linux-aarch64-3.8/cffi
    copying cffi/_cffi_include.h -> build/lib.linux-aarch64-3.8/cffi
    copying cffi/parse_c_type.h -> build/lib.linux-aarch64-3.8/cffi
    copying cffi/_embedding.h -> build/lib.linux-aarch64-3.8/cffi
    copying cffi/_cffi_errors.h -> build/lib.linux-aarch64-3.8/cffi
    running build_ext
    building '_cffi_backend' extension
    creating build/temp.linux-aarch64-3.8
    creating build/temp.linux-aarch64-3.8/c
    gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -g -fno-semantic-interposition -g -fno-semantic-interposition -g -fno-semantic-interposition -DTHREAD_STACK_SIZE=0x100000 -fPIC -I/usr/include/ffi -I/usr/include/libffi -I/usr/include/python3.8 -c c/_cffi_backend.c -o build/temp.linux-aarch64-3.8/c/_cffi_backend.o
    unable to execute 'gcc': No such file or directory
    error: command 'gcc' failed with exit status 1
    ----------------------------------------
    ERROR: Failed building wheel for cffi
    Running setup.py clean for cffi
  Failed to build cffi
  Installing collected packages: setuptools, wheel, pycparser, cffi
      Running setup.py install for cffi: started
      Running setup.py install for cffi: finished with status 'error'
      ERROR: Command errored out with exit status 1:
       command: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-avyx4ntk/cffi/setup.py'"'"'; __file__='"'"'/tmp/pip-install-avyx4ntk/cffi/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-aeatxisk/install-record.txt --single-version-externally-managed --prefix /tmp/pip-build-env-vg6dkv2y/overlay --compile --install-headers /tmp/pip-build-env-vg6dkv2y/overlay/include/python3.8/cffi
           cwd: /tmp/pip-install-avyx4ntk/cffi/
      Complete output (68 lines):
      Package libffi was not found in the pkg-config search path.
      Perhaps you should add the directory containing `libffi.pc'
      to the PKG_CONFIG_PATH environment variable
      Package 'libffi', required by 'virtual:world', not found
      Package libffi was not found in the pkg-config search path.
      Perhaps you should add the directory containing `libffi.pc'
      to the PKG_CONFIG_PATH environment variable
      Package 'libffi', required by 'virtual:world', not found
      Package libffi was not found in the pkg-config search path.
      Perhaps you should add the directory containing `libffi.pc'
      to the PKG_CONFIG_PATH environment variable
      Package 'libffi', required by 'virtual:world', not found
      Package libffi was not found in the pkg-config search path.
      Perhaps you should add the directory containing `libffi.pc'
      to the PKG_CONFIG_PATH environment variable
      Package 'libffi', required by 'virtual:world', not found
      Package libffi was not found in the pkg-config search path.
      Perhaps you should add the directory containing `libffi.pc'
      to the PKG_CONFIG_PATH environment variable
      Package 'libffi', required by 'virtual:world', not found
      unable to execute 'gcc': No such file or directory
      unable to execute 'gcc': No such file or directory
  
          No working compiler found, or bogus compiler options passed to
          the compiler from Python's standard "distutils" module.  See
          the error messages above.  Likely, the problem is not related
          to CFFI but generic to the setup.py of any Python package that
          tries to compile C code.  (Hints: on OS/X 10.8, for errors about
          -mno-fused-madd see http://stackoverflow.com/questions/22313407/
          Otherwise, see https://wiki.python.org/moin/CompLangPython or
          the IRC channel #python on irc.freenode.net.)
  
          Trying to continue anyway.  If you are trying to install CFFI from
          a build done in a different context, you can ignore this warning.
  
      running install
      running build
      running build_py
      creating build
      creating build/lib.linux-aarch64-3.8
      creating build/lib.linux-aarch64-3.8/cffi
      copying cffi/setuptools_ext.py -> build/lib.linux-aarch64-3.8/cffi
      copying cffi/api.py -> build/lib.linux-aarch64-3.8/cffi
      copying cffi/pkgconfig.py -> build/lib.linux-aarch64-3.8/cffi
      copying cffi/vengine_cpy.py -> build/lib.linux-aarch64-3.8/cffi
      copying cffi/cparser.py -> build/lib.linux-aarch64-3.8/cffi
      copying cffi/__init__.py -> build/lib.linux-aarch64-3.8/cffi
      copying cffi/verifier.py -> build/lib.linux-aarch64-3.8/cffi
      copying cffi/lock.py -> build/lib.linux-aarch64-3.8/cffi
      copying cffi/backend_ctypes.py -> build/lib.linux-aarch64-3.8/cffi
      copying cffi/error.py -> build/lib.linux-aarch64-3.8/cffi
      copying cffi/vengine_gen.py -> build/lib.linux-aarch64-3.8/cffi
      copying cffi/recompiler.py -> build/lib.linux-aarch64-3.8/cffi
      copying cffi/model.py -> build/lib.linux-aarch64-3.8/cffi
      copying cffi/cffi_opcode.py -> build/lib.linux-aarch64-3.8/cffi
      copying cffi/ffiplatform.py -> build/lib.linux-aarch64-3.8/cffi
      copying cffi/commontypes.py -> build/lib.linux-aarch64-3.8/cffi
      copying cffi/_cffi_include.h -> build/lib.linux-aarch64-3.8/cffi
      copying cffi/parse_c_type.h -> build/lib.linux-aarch64-3.8/cffi
      copying cffi/_embedding.h -> build/lib.linux-aarch64-3.8/cffi
      copying cffi/_cffi_errors.h -> build/lib.linux-aarch64-3.8/cffi
      running build_ext
      building '_cffi_backend' extension
      creating build/temp.linux-aarch64-3.8
      creating build/temp.linux-aarch64-3.8/c
      gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -g -fno-semantic-interposition -g -fno-semantic-interposition -g -fno-semantic-interposition -DTHREAD_STACK_SIZE=0x100000 -fPIC -I/usr/include/ffi -I/usr/include/libffi -I/usr/include/python3.8 -c c/_cffi_backend.c -o build/temp.linux-aarch64-3.8/c/_cffi_backend.o
      unable to execute 'gcc': No such file or directory
      error: command 'gcc' failed with exit status 1
      ----------------------------------------
  ERROR: Command errored out with exit status 1: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-avyx4ntk/cffi/setup.py'"'"'; __file__='"'"'/tmp/pip-install-avyx4ntk/cffi/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-aeatxisk/install-record.txt --single-version-externally-managed --prefix /tmp/pip-build-env-vg6dkv2y/overlay --compile --install-headers /tmp/pip-build-env-vg6dkv2y/overlay/include/python3.8/cffi Check the logs for full command output.
  WARNING: You are using pip version 20.2.4; however, version 21.1.2 is available.
  You should consider upgrading via the '/usr/bin/python3 -m pip install --upgrade pip' command.
  ----------------------------------------
ERROR: Command errored out with exit status 1: /usr/bin/python3 /usr/lib/python3.8/site-packages/pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-vg6dkv2y/overlay --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- setuptools wheel 'cffi>=1.4.1; python_implementation != '"'"'PyPy'"'"'' Check the logs for full command output.
WARNING: You are using pip version 20.2.4; however, version 21.1.2 is available.
You should consider upgrading via the '/usr/bin/python3 -m pip install --upgrade pip' command.
2021-06-21 22:15:48 ERROR (MainThread) [homeassistant.config] Package mobile_app setup failed. Integration mobile_app Requirements for mobile_app not found: ['PyNaCl==1.3.0']. (See /config/integrations/mobile_app.yaml:7). 
2021-06-21 22:15:52 WARNING (MainThread) [homeassistant.components.lovelace] Lovelace is running in storage mode. Define resources via user interface
2021-06-21 22:16:04 WARNING (MainThread) [homeassistant.setup] Setup of zone is taking over 10 seconds.
2021-06-21 22:16:04 WARNING (MainThread) [homeassistant.setup] Setup of input_boolean is taking over 10 seconds.
```
Removing my custom-init-scripts which adds python2 didn't resolve the issue.
Also tried tested the config on another raspberry pi 3b+ and ran into same issue.

**Comment by aptalca** [comment]
Thanks, we're aware of the issue with the latest build and are looking into it. You can pull the previous version until it's fixed.

**Comment by kavemang** [comment]
I have the same issue, but after my mobile client tried to connect I can no longer seem to connect by any method

**Comment by Jayrayme** [comment]
Same issue, i rolled back to the previous version until it's fixed.

**Comment by vcao** [comment]
I've rolled back to 2021.6.0-ls24 for the time being and it works. If you're in unraid change your repo for the time being to:
ghcr.io/linuxserver/homeassistant:2021.6.0-ls24

Once a permanent fix is working then you can change it back to:
ghcr.io/linuxserver/homeassistant:latest

**Comment by MunkeyBalls** [comment]
@aptalca It seems this has been already fixed, I've pulled the latest image (2021.7.0) and so far everything is working again. 

Thanks for your work, feel free to close 👍
